Cannot re-export type from generic components

### Vue - Official extension or vue-tsc version

vue-tsc - Version 5.8.3

### VSCode version

\-

### Vue version

3.5.16

### TypeScript version

5.8.3

### System Info

```shell
stackblitz
```

### package.json dependencies

```json
{
  "type": "module",
  "devDependencies": {
    "@vue/tsconfig": "^0.7.0",
    "typescript": "^5.8.3",
    "vue": "^3.5.16",
    "vue-tsc": "^2.2.10"
  }
}
```

### Steps to reproduce

See link below.

Run `vue-tsc --noEmit`.

Copying for reference:

Don't work:

```vue
<script setup lang="ts" generic="T">
import type { Props } from './bar.js';

export type { Props };

defineProps<Props<T>>();
</script>

<template></template>
```

```vue
<script setup lang="ts" generic="T">
import type { Props } from './bar.js';

export type { Props } from './bar.js';

defineProps<Props<T>>();
</script>

<template></template>
```

Work:

```vue
<script setup lang="ts" generic="T">
import type { Props as _Props } from './bar.js';

export type Props<T> = _Props<T>;

defineProps<Props<T>>();
</script>

<template></template>
```

```vue
<script setup lang="ts">
import type { Props } from './bar.js';

export type { Props };

defineProps<Props<any>>();
</script>

<template></template>
```

### What is expected?

No errors.

### What is actually happening?

Errors:

```sh
Foo1.vue:4:1 - error TS1233: An export declaration can only be used at the top level of a namespace or module.

4 export type { Props };
  ~~~~~~

Foo1.vue:4:15 - error TS2303: Circular definition of import alias 'Props'.

4 export type { Props };
                ~~~~~

Foo2.vue:4:1 - error TS1233: An export declaration can only be used at the top level of a namespace or module.

4 export type { Props } from './bar.js';
  ~~~~~~


Found 3 errors in 2 files.

Errors  Files
     2  Foo1.vue:4
     1  Foo2.vue:4
```

### Link to minimal reproduction

https://stackblitz.com/edit/node-z86z5kqe?file=Foo1.vue,Foo2.vue,Foo3.vue,Foo4.vue

### Any additional comments?

![Image](https://github.com/user-attachments/assets/879cc2cd-5de2-414e-a0b3-502ca41f8ce8)

--- a/packages/language-core/lib/parsers/scriptSetupRanges.ts
+++ b/packages/language-core/lib/parsers/scriptSetupRanges.ts
@@ -100,27 +100,33 @@ export function parseScriptSetupRanges(
 	let importSectionEndOffset = 0;
 
 	ts.forEachChild(ast, node => {
-		const isTypeExport =
-			(ts.isTypeAliasDeclaration(node) || ts.isInterfaceDeclaration(node))
-			&& node.modifiers?.some(mod => mod.kind === ts.SyntaxKind.ExportKeyword);
 		if (
-			!foundNonImportExportNode
-			&& !ts.isImportDeclaration(node)
-			&& !isTypeExport
-			&& !ts.isEmptyStatement(node)
+			foundNonImportExportNode
+			|| ts.isImportDeclaration(node)
+			|| ts.isExportDeclaration(node)
+			|| ts.isEmptyStatement(node)
 			// fix https://github.com/vuejs/language-tools/issues/1223
-			&& !ts.isImportEqualsDeclaration(node)
+			|| ts.isImportEqualsDeclaration(node)
 		) {
-			const commentRanges = ts.getLeadingCommentRanges(text, node.pos);
-			if (commentRanges?.length) {
-				const commentRange = commentRanges.sort((a, b) => a.pos - b.pos)[0];
-				importSectionEndOffset = commentRange.pos;
-			}
-			else {
-				importSectionEndOffset = getStartEnd(ts, node, ast).start;
-			}
-			foundNonImportExportNode = true;
+			return;
 		}
+
+		if (
+			(ts.isTypeAliasDeclaration(node) || ts.isInterfaceDeclaration(node))
+			&& node.modifiers?.some(mod => mod.kind === ts.SyntaxKind.ExportKeyword)
+		) {
+			return;
+		}
+
+		const commentRanges = ts.getLeadingCommentRanges(text, node.pos);
+		if (commentRanges?.length) {
+			const commentRange = commentRanges.sort((a, b) => a.pos - b.pos)[0];
+			importSectionEndOffset = commentRange.pos;
+		}
+		else {
+			importSectionEndOffset = getStartEnd(ts, node, ast).start;
+		}
+		foundNonImportExportNode = true;
 	});
 	ts.forEachChild(ast, node => visitNode(node, [ast]));
 

